docs/en/api/FS-API.md

```diff
@@ -187,23 +212,40 @@ Users can override the default policy class in the
 
-    Returns the local worker first, and if it does not have enough capacity of a block,
-    randomly picks a worker from the active workers list. This is the default policy.
+  > If no worker meets the demands, return local host. USER_FILE_WRITE_AVOID_EVICTION_POLICY_RESERVED_BYTES
+  > is used to reserve some space of the worker to store the block, for the values
+  > mCapacityBytes minus mUsedBytes is not the available bytes.
 
-* **MostAvailableFirstPolicy (alluxio.client.block.policy.MostAvailableFirstPolicy)**
+* [MostAvailableFirstPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/MostAvailableFirstPolicy.html)
 
-    Returns the worker with the most available bytes.
+  > A policy that returns the worker with the most available bytes. The policy returns null if no worker is qualified.
 
-* **RoundRobinPolicy (alluxio.client.block.policy.RoundRobinPolicy)**
+* [RoundRobinPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/RoundRobinPolicy.html)
 
-    Chooses the worker for the next block in a round-robin manner and skips workers that do not have
-    enough capacity.
+  > A policy that chooses the worker for the next block in a round-robin manner
+  > and skips workers that do not have enough space. The policy returns null if no worker can be found.
 
-* **SpecificHostPolicy (alluxio.client.block.policy.SpecificHostPolicy)**
+* [SpecificHostPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/SpecificHostPolicy.html)
 
-    Returns a worker with the specified host name. This policy cannot be set as default policy.
+  > Always returns a worker with the specified hostname. Returns null if no active worker on that hostname found.
+
+* [DeterministicHashPolicy](https://docs.alluxio.io/os/javadoc/{{site.ALLUXIO_MAJOR_VERSION}}/alluxio/client/block/policy/DeterministicHashPolicy.html)
+
+  > This policy maps blockId to several deterministic Alluxio workers.
+  > The number of workers a block can be mapped to can be passed through the constructor.
+  > The default is 1. It skips the workers that do not have enough capacity to hold the block.
+  > Note that the hash function relies on the number of workers in the cluster, so if the number of workers changes, the workers chosen by the policy for a given block will likely change. This policy is useful for limiting the amount of replication that occurs when reading blocks from the UFS with high concurrency. With 30 workers and 100 remote clients reading the same block concurrently, the replication level for the block would get close to 30 as each workers reads and caches the block for one or more clients. If the clients use DeterministicHashPolicy with 3 shards, the 100 clients will split their reads between just 3 workers, so that the replication level for the block will be only 3 when the data is first loaded.
```

Comment:
👍  the explanation makes a lot of sense even though i had no idea what this does.

split this up into separate lines for editability

`Note that the hash function relies on the number of workers in the cluster, so if the number of workers changes, the workers chosen by the policy for a given block will likely change.` feels like it belongs at the end as a warning, after explaining how the policy internals operate.